### PR TITLE
GH-1909 Add validation for chef_server_url

### PR DESCRIPTION
**Pull request by skmichaelson**

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by thommay** [comment]
Hi @skmichaelson, to progress further with this PR we'll need to ask you to sign a CLA.

**Comment by skmichaelson** [comment]
@thommay, I signed the ICLA on August 20th. My Chef Supermarket profile says that I'm authorized to contribute:

supermarket

Is there something else I forgot to do?

**Comment by thommay** [comment]
Ah, the email address that you used to commit the changes isn't linked to your github account. Can you either add the email address to your github account ( https://help.github.com/articles/adding-an-email-address-to-your-github-account/ ) or follow https://help.github.com/articles/setting-your-email-in-git/ to set your email address locally; you'll then need to change your previous commits. I'd achieve this by doing a rebase and squash - https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history is a good guide to this. Thanks!

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @veetow
* @juliegund

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by thommay** [comment]
:+1:

File: chef-config/lib/chef-config/config.rb [hunk 1]
```diff
@@ -76,12 +76,21 @@ def self.add_event_logger(logger)
...
+    end
     # Override the config dispatch to set the value of multiple server options simultaneously
     #
     # === Parameters
     # url<String>:: String to be set for all of the chef-server-api URL's
     #
-    configurable(:chef_server_url).writes_value { |url| url.to_s.strip }
+    configurable(:chef_server_url).writes_value do |uri|
+      unless is_valid_url? uri
+        raise ConfigurationError, "#{url} is an invalid chef_server_url."
```
Comment (btm) [inline chef-config/lib/chef-config/config.rb]: s/url/uri/. I'll fix this on merge. :+1:

**Pull request closed without merging**